bucket.go

```diff
@@ -20,6 +20,8 @@ type Bucket interface {
 	GetName() string
 	Get(k string, rv interface{}) error
 	GetRaw(k string) ([]byte, error)
+	Gets(k string, rv interface{}, caso *uint64) error
```

Comment:
Agreed - I'd been leaning towards the gocb approach, but was wary of the amount of refactoring needed in SG to switch to the new Get.  I feel like I should do that refactoring in sync-gateway/master - let me know if you disagree.